test/unit/stats.js

```diff
@@ -0,0 +1,793 @@
+        }
+      };
+
+      var promise = stats.getStats({});
+      var statCount;
+
+      promise.then(function(data) {
+        statCount = data.length;
+        assert.deepEqual(data, [fakeStat, fakeStat2, fakeStat3,
+          fakeStat4, fakeStat5, fakeStat6]);
+
+        return stats.createStats(req);
+      })
+      .then(function() {
+        return stats.getStats({});
+      })
+      .then(function(data) {
+        assert.equal(err.message,
+        'Could not post due to missing fields');
+        assert.equal(err.status, 400);
+        assert.lengthOf(data, statCount);
+        assert.deepEqual([fakeStat, fakeStat2, fakeStat3,
+          fakeStat4, fakeStat5, fakeStat6]);
+        done();
+      });
+    });
+
+    // check error is thrown if required field 'event_id' is missing in request
+    xit('attempting to post stats with missing field event_id will result in an error', function(done) {
+      var req = {
+        body: {
+          student_id: 1,
+          pacer: 5,
+          height: 5,
+          weight: 5,
+        }
+      };
+
+      var promise = stats.getStats({});
+      var statCount;
+
+      promise.then(function(data) {
+        statCount = data.length;
+        assert.deepEqual(data, [fakeStat, fakeStat2, fakeStat3,
+          fakeStat4, fakeStat5, fakeStat6]);
+
+        return stats.createStats(req);
+      })
+      .then(function() {
+        return stats.getStats({});
+      })
+      .then(function(data) {
+        assert.equal(err.message,
+        'Could not post due to missing fields');
+        assert.equal(err.status, 400);
+        assert.lengthOf(data, statCount);
+        assert.deepEqual([fakeStat, fakeStat2, fakeStat3,
+          fakeStat4, fakeStat5, fakeStat6]);
+        done();
+      });
+    });
+
+    // check error is thrown if required field 'student_id' is missing in request
+    xit('attempting to post stats with missing field student_id will result in an error', function(done) {
+      var req = {
+        body: {
+          pacer: 5,
+          event_id: 1,
+          height: 5,
+          weight: 5,
+        }
+      };
+
+      var promise = stats.getStats({});
+      var statCount;
+
+      promise.then(function(data) {
+        statCount = data.length;
+        assert.deepEqual(data, [fakeStat, fakeStat2, fakeStat3,
+          fakeStat4, fakeStat5, fakeStat6]);
+
+        return stats.createStats(req);
+      })
+      .then(function() {
+        return stats.getStats({});
+      })
+      .then(function(data) {
+        assert.equal(err.message,
+        'Could not post due to missing fields');
+        assert.equal(err.status, 400);
+        assert.lengthOf(data, statCount);
+        assert.deepEqual([fakeStat, fakeStat2, fakeStat3,
+          fakeStat4, fakeStat5, fakeStat6]);
+        done();
+      });
+    });
+
+    // check that POST doesn't happen if field is repeated
+    xit('attempting to post stats with duplicate fields will result in an error', function(done) {
```

Comment:
I found out Javascript just eliminates the first instance of the field, so this test can never actually work. Delete it.